compiler/rustc_mir_build/src/thir/constant.rs

```diff
@@ -8,6 +8,66 @@ use crate::build::parse_float_into_scalar;
 pub(crate) fn lit_to_const<'tcx>(
```

Comment:
naming: we could call this `lit_to_const_eager` or something, to help make it clear that it's trying to eagerly convert lits when possible, but will hold off from doing so if not possible (i.e. we don't have the right type info).